FWCore/ParameterSet/python/Types.py

```diff
@@ -659,7 +659,15 @@ def _setValues(self,moduleLabel,productInstanceLabel='',processName=''):
         self.__moduleLabel = moduleLabel
         self.__productInstance = productInstanceLabel
         self.__processName=processName
-        if -1 != moduleLabel.find(":"):
+        if isinstance(moduleLabel, tuple):
```

Comment:
We should also support lists

Reply:
Our messages crossed, I'll add the list support then.

Reply:
I was trying to see if I could come up with our needs based on needed operations rather than type. What we require is

1. support `len` (i.e. `__len__`)
1. supports indexed lookup using an integer and the result of the lookup returns a string (i.e. `__getitem__`)

if that is true then everything is fine

Reply:
I thought that too, but `string` itself satisfies both.

Reply:
Yes, but now you are checking for container and assuming everything is a string (which should throw if not). Might be better to reverse the logic and check for string and then assume the type is a usable container?

Reply:
Makes sense, thanks. We seem to be using that pattern (check if string, else use as a container) already.